cpp/src/cylon/arrow/arrow_types.hpp

```diff
@@ -28,11 +28,18 @@ namespace tarrow {
  * @param tType the cylon type
  * @return corresponding arrow type
  */
-std::shared_ptr<arrow::DataType> convertToArrowType(std::shared_ptr<DataType> tType,
+std::shared_ptr<arrow::DataType> convertToArrowType(const std::shared_ptr<DataType> &tType,
```

Comment:
WE should validate width, percision when using. otherwise it will set to -1 which not sensible